### PR TITLE
e-2197-qa-accross-the-map-on-va-chez-le-voisin

### PR DESCRIPTION
**Pull request by haydencleary**

### CODE REVIEW

File: core/kernel/money/money-adapter.ts [hunk 1]
```diff
@@ -108,14 +110,16 @@ export class MoneyAdapter implements MoneyFacadePort {
...
 
     // Need this to have fixed decimal places for compact notation
     const maximumFractionDigits = notation === "compact" ? 1 : decimals;
 
     return new Intl.NumberFormat(locale, {
       maximumFractionDigits,
       notation,
+      maximumSignificantDigits: this.maximumSignificantDigits,
+      ...restOptions,
     })
```
Comment (alexbeno) [inline core/kernel/money/money-adapter.ts:123]:
```suggestion
    // Need this to have fixed decimal places for compact notation
    const maximumFractionDigits = options.notation === "compact" ? 1 : decimals;

    return new Intl.NumberFormat(locale, {
      maximumFractionDigits,
      maximumSignificantDigits: this.maximumSignificantDigits,
      ...options,
    })
```

Comment (alexbeno) [reply core/kernel/money/money-adapter.ts:123]: ?

File: shared/features/amount-selector/amount-selector.tsx [hunk 2]
```diff
@@ -81,16 +93,16 @@ export function AmountSelector({
...
+            "text-2xl": formattedAmount.length < 18,
+            "text-3xl": formattedAmount.length < 14,
+            "text-4xl": formattedAmount.length < 11,
           })}
         >
           <input
             ref={inputRef}
             type="text"
-            style={{ width: Math.min(Math.max(amount.length, 2), 50) + "ch" }}
+            style={{ width: Math.min(Math.max(formattedAmount.length, 2), 50) + "ch" }}
```
Comment (alexbeno) [inline shared/features/amount-selector/amount-selector.tsx:105]: you can extract this in a const / memo

File: shared/features/amount-selector/amount-selector.tsx [hunk 3]
```diff
@@ -61,10 +61,22 @@ export function AmountSelector({
...
         value = value.slice(1);
       }
 
+      if (
+        (value.includes(".") && value.length > maximumSignificantDigits + 1) ||
+        value.length > maximumSignificantDigits
+      ) {
+        return;
+      }
+
```
Comment (alexbeno) [inline shared/features/amount-selector/amount-selector.tsx:79]:
```suggestion
      if (
        value.length > maximumSignificantDigits
      ) {
        return;
      }
      
            if (
        value.includes(".") && value.length > maximumSignificantDigits + 1
      ) {
        return;
      }
      
      

```

**Review by pixelfact — APPROVED** [review]

**Pull request merged**